api/external/data_lifecycle/data_lifecycle.proto

```diff
@@ -0,0 +1,294 @@
+
+	//
+	// Compliance
+	//
+
+	// GetComplianceStatus returns the compliance job scheduler status
+	rpc GetComplianceStatus(GetComplianceStatusRequest) returns (GetComplianceStatusResponse) {
+		option (google.api.http).get = "/data-lifecycle/compliance/status";
+		option (chef.automate.api.policy).resource = "datalifecyle:compliance:status";
+		option (chef.automate.api.policy).action = "read";
+		option (chef.automate.api.iam.policy).resource = "dataLifecycle:compliance:status";
+		option (chef.automate.api.iam.policy).action = "dataLifecycle:compliance:get";
+	};
+
+	// RunCompliance runs the compliance data lifecycle jobs
+	rpc RunCompliance(RunComplianceRequest) returns (RunComplianceResponse) {
+		option (google.api.http) = {
+			post: "/data-lifecycle/compliance/run";
+			body: "*";
+		};
+		option (chef.automate.api.policy).resource = "dataLifecycle:compliance:run";
+		option (chef.automate.api.policy).action = "create";
+		option (chef.automate.api.iam.policy).resource = "dataLifecycle:compliance:run";
+		option (chef.automate.api.iam.policy).action = "dataLifecycle:compliance:create";
+	};
+
+	// SetComplianceConfig configures the compliance data lifecycle scheduler and jobs
+	rpc SetComplianceConfig(SetComplianceConfigRequest) returns (SetComplianceConfigResponse) {
+		option (google.api.http) = {
+			put: "/data-lifecycle/compliance/config";
+			body: "*"
+		};
+		option (chef.automate.api.policy).resource = "dataLifecycle:compliance:config";
+		option (chef.automate.api.policy).action = "update";
+		option (chef.automate.api.iam.policy).resource = "dataLifecycle:compliance:config";
+		option (chef.automate.api.iam.policy).action = "dataLifecycle:compliance:update";
+	};
+}
+
+// JobStatus presents the current configuration of job, when it will be executed,
+// and details about it's most recent execution.
+message JobStatus {
+	string name = 1;
+	bool disabled = 2;
+	string recurrence = 3;
+	string threshold = 4;
+	PurgePolicies purge_policies = 5;
+	google.protobuf.Duration last_elapsed = 6;
+	google.protobuf.Timestamp next_due_at = 7;
+	google.protobuf.Timestamp last_enqueued_at = 8;
+	google.protobuf.Timestamp last_started_at = 9;
+	google.protobuf.Timestamp last_ended_at = 10;
+}
+
+// PurgePolicies are data lifecycle purge policies
+message PurgePolicies {
+	repeated chef.automate.infra.data_lifecycle.api.EsPolicy elasticsearch = 1;
+	repeated chef.automate.infra.data_lifecycle.api.PgPolicy postgres = 2;
+}
+
+// PurgePolicyUpdate is purge policy configuration update
+message PurgePolicyUpdate {
+	repeated chef.automate.infra.data_lifecycle.api.EsPolicyUpdate elasticsearch = 1;
+	repeated chef.automate.infra.data_lifecycle.api.PgPolicyUpdate postgres = 2;
+}
+
+// JobSettings are a job configuration setting update
+message JobSettings {
+	string name =1;
+	bool disabled = 2;
+	string recurrence = 3;
+	string threshold = 4;
+	PurgePolicyUpdate purge_policies = 5;
+}
+
+message GetStatusRequest { }
+message GetStatusResponse {
```

Comment:
It seems all of the messages below are the same. Would it make sense to just have 1 message type and use a map[MessageType] in the Getters/Setters

Reply:
a related thing (not sure how possible it is given the authz stuff): we probably could just have just 1 of each of Set/Get/Status RPCs.
https://github.com/googleapis/googleapis/blob/master/google/api/http.proto#L73 shows off some of that

Reply:
Way back in the day we did that but if you ever want to change the result you're super boxed in and have to change it for everything. I don't particularly like having independent messages everywhere because in most cases YAGNI, but alas, if we do it's certainly nice.